python/cudf/cudf/core/series.py

```diff
@@ -3544,7 +3562,7 @@ def encode(cat):
             elif np.issubdtype(type(cat), np.floating) and np.isnan(cat):
                 return self.__class__(libcudf.unary.is_nan(self._column))
             else:
-                return (self == cat).fillna(False)
+                raise ValueError("Unsupported cat type.")
 
         return [encode(cat).astype(dtype) for cat in cats]
```

Comment:
That's a very good catch! The code after `encode` is not needed anymore because they are unreachable at this point. `cats` will not be None because of the preprocessing above.